code/modules/heavy_vehicle/equipment/medical.dm

```diff
@@ -263,6 +264,76 @@
 	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
 	restricted_software = list(MECH_SOFTWARE_MEDICAL)
 
+/obj/item/device/healthanalyzer/mech //Used to set up the full body scan feature
+	var/obj/machinery/body_scanconsole/internal_bodyscanner = null
+	var/fullScan = FALSE //Toggle whether to do full or basic scan
+	
+/obj/item/device/healthanalyzer/mech/Initialize()
+	. = ..()
+	if(!internal_bodyscanner)
+		var/obj/machinery/body_scanconsole/S = new (src)
+		S.forceMove(src)
+		S.use_power = FALSE
+		internal_bodyscanner = S
+
+/obj/item/mecha_equipment/mounted_system/medanalyzer/CtrlClick(mob/user)
+	var/obj/item/device/healthanalyzer/mech/HA = holding
+	if(istype(HA))
+		HA.fullScan = !HA.fullScan
+		to_chat(user, SPAN_NOTICE("You switch to \the [src]'s [HA.fullScan ? "full body" : "basic"] scan mode."))
+
 /obj/item/device/healthanalyzer/mech/attack(mob/living/M, var/mob/living/heavy_vehicle/user)
-	for(var/mob/pilot in user.pilots)
-		health_scan_mob(M, pilot, TRUE, TRUE)
+	if(!fullScan)
+		for(var/mob/pilot in user.pilots)
+			health_scan_mob(M, pilot, TRUE, TRUE)
+	else
+		user.visible_message("<b>[user]</b> starts scanning \the [M] with \the [src].", SPAN_NOTICE("You start scanning \the [M] with \the [src]."))
+		if(do_after(user, 7 SECONDS, TRUE))
+			print_scan(M, user)
+			add_fingerprint(user)
+
+/obj/item/device/healthanalyzer/mech/proc/print_scan(var/mob/M, var/mob/living/user)
+	var/obj/item/paper/medscan/R = new(user.loc)
+	R.color = "#eeffe8"
+	R.set_content_unsafe("Scan ([M.name])", internal_bodyscanner.format_occupant_data(get_medical_data(M)))
+
+	if(ishuman(user) && !(user.l_hand && user.r_hand))
+		user.put_in_hands(R)
+	user.visible_message("\The [src] spits out a piece of paper.")
+
+/obj/item/device/healthanalyzer/mech/proc/get_medical_data(var/mob/living/carbon/human/H)
```

Comment:
This looks like horrid copypaste from bodyscanner code. Why does this have to be copypasted?

Reply:
Discussed in discord. Tears were shed.